### PR TITLE
Persistence features for Runtime

### PR DESCRIPTION
**Pull request by Incoming5643**
- Runtime now keeps track of all cats in the world, at the start of the next round any cats that survived the previous one will spawn with Runtime in the CMO's office.
- After 9 deaths Runtime will be mysteriously missing from the next station (but only for that round). This also clears out any family Runtime might have.

---

A note on mechanics: Yes this allows for potentially infinite cats, but in reality this is not a practical problem because:
1. Runtime and Proc can have kittens, and kittens are saved BUT they can only have three. This is true even if they start with three kittens they didn't actually have. It's technically possible to get more kittens by repeatedly removing new kittens from the cat breeding chamber, but that's a lot of work just to catsplosion the CMO next round.
2. The petsplosion wizard event can exacerbate a cat situation, but Runtime won't carry over other versions of herself, and since Runtime is the only female cat in the whole dang game this puts a bottleneck on kittens that should keep things from going nuts. 
3. It only takes one successful nuke round to genocide the cats.
4. :cat2:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -60,6 +60,60 @@
...
+		lives = 9
+
+	if(lives <= 0)
+		lives = 10 //Lowers to 9 in Write_Memory
+		Write_Memory(1)
+		qdel(src)
+
+	for(var/cat_type in family)
+		if(family[cat_type] > 0)
+			for(var/i = 0, i < family[cat_type], i++)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: ```for(var/i in 0 to family[cat_type]-1)```
or
```for(var/i in 1 to family[cat_type])```

**Comment by PKPenguin321** [comment]
You should add a line to it's examine text saying how many lives it has.
something like,
"If cats truly have nine lives, then this looks like it must have died [once/twice/three times/four times etc.] already."

**Comment by Zilenan91** [comment]
What about kittens in the pet boxes in mining artifact rooms, will those go to the CMO's office

**Comment by Incoming5643** [comment]
Animals in cargo crates don't actually spawn until they're opened, so they won't be saved.

The holodeck cats are definitely an open question though

**Comment by Zilenan91** [comment]
It's probably fine because I can call the CMO a crazy cat lady

**Comment by KorPhaeron** [comment]
lmao2cat

**Comment by Niknakflak** [comment]
You are driving this into the ground tbh
It made sense with poly but now its just meh.
It's cat PR so :+1: but eh

**Comment by Incoming5643** [comment]
I want to give every pet a thing so no one feels left out.

Except maybe that fox, fuck him.

**Comment by Razharas** [comment]
WHen will we be able to give runtime an antag role?
@Incoming5643

**Comment by bgobandit** [comment]
I support cat functionality

**Comment by Aloraydrel** [comment]
A FELINE PAIR!!

**Comment by bgobandit** [comment]
I just got the pun in the branch name

you devil you

**Comment by duncathan** [comment]
:+1:

**Comment by Iamgoofball** [comment]
@Incoming5643 

Cats should follow around the Janicart if the keys are in and it's being driven around.

**Comment by Shadowlight213** [comment]
@incoming Lamarr expansion when?

**Comment by Incoming5643** [comment]
:cat2: Should be good to go now :cat2:

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -54,12 +54,68 @@
 //RUNTIME IS ALIVE! SQUEEEEEEEE~
 /mob/living/simple_animal/pet/cat/Runtime
 	name = "Runtime"
-	desc = "GCAT"
+	desc = "Tends to show up in the strangest of places."
```
Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: noooo put it back put it back ;_;

**Comment by Incoming5643** [comment]
just for you @duncathan

**Comment by RemieRichards** [comment]
I'm disappointed you tried to remove the GCAT description.
its literally the smartest joke we have.

**Comment by Incoming5643** [comment]
Why did you have to say that now I gotta code dive descriptions to make sure your right.

**Pull request merged**